code/game/machinery/computer/security.dm

```diff
@@ -319,237 +200,228 @@
+  *
+  * Arguments:
+  * * action - The action passed by tgui
+  * * params - The params passed by tgui
+  */
+/obj/machinery/computer/secure_data/proc/tgui_act_modal(action, list/params)
+	. = TRUE
+	var/id = params["id"]
+	var/list/arguments = istext(params["arguments"]) ? json_decode(params["arguments"]) : params["arguments"]
+	switch(tgui_modal_act(src, action, params))
+		if(TGUI_MODAL_OPEN)
+			switch(id)
+				if("edit")
+					var/field = arguments["field"]
+					if(!length(field) || !field_edit_questions[field])
+						return
+					var/question = field_edit_questions[field]
+					var/choices = field_edit_choices[field]
+					if(length(choices))
+						tgui_modal_choice(src, id, question, arguments = arguments, value = arguments["value"], choices = choices)
+					else
+						tgui_modal_input(src, id, question, arguments = arguments, value = arguments["value"])
+				if("comment_add")
+					tgui_modal_input(src, id, "Please enter your message:")
+				if("print_cell_log")
+					if(is_printing)
+						return
+					if(!length(GLOB?.cell_logs))
+						set_temp("There are no cell logs available to print.")
+						return
+					var/list/choices = list()
+					var/list/already_in[0]
+					for(var/obj/item/paper/P in GLOB.cell_logs)
+						if(already_in[P.name])
+							continue
+						choices += P.name
+						already_in[P.name] = TRUE
+					tgui_modal_choice(src, id, "Please select the cell log you would like printed:", choices = choices)
 				else
-					P.info += "<B>General Record Lost!</B><BR>"
-				if(istype(active2, /datum/data/record) && GLOB.data_core.security.Find(active2))
-					P.info += {"<BR>\n<CENTER><B>Security Data</B></CENTER>
-					<BR>\nCriminal Status: [active2.fields["criminal"]]<BR>\n
-					<BR>\nMinor Crimes: [active2.fields["mi_crim"]]
-					<BR>\nDetails: [active2.fields["mi_crim_d"]]<BR>\n
-					<BR>\nMajor Crimes: [active2.fields["ma_crim"]]
-					<BR>\nDetails: [active2.fields["ma_crim_d"]]<BR>\n
-					<BR>\nImportant Notes:
-					<BR>\n\t[active2.fields["notes"]]<BR>\n<BR>\n<CENTER><B>Comments/Log</B></CENTER><BR>"}
-					for(var/c in active2.fields["comments"])
-						P.info += "[c]<BR>"
+					return FALSE
+		if(TGUI_MODAL_ANSWER)
+			var/answer = params["answer"]
+			switch(id)
+				if("edit")
+					var/field = arguments["field"]
+					if(!length(field) || !field_edit_questions[field])
+						return
+					var/list/choices = field_edit_choices[field]
+					if(length(choices) && !(answer in choices))
+						return
+
+					if(field == "age")
+						var/new_age = text2num(answer)
+						if(new_age < AGE_MIN || new_age > AGE_MAX)
+							set_temp("Invalid age. It must be between [AGE_MIN] and [AGE_MAX].", "danger")
+							return
+						answer = new_age
+					else if(field == "criminal")
+						var/text = "Please enter a reason for the status change to [answer]:"
+						if(answer == SEC_RECORD_STATUS_EXECUTE)
+							text = "Please explain why they are being executed. Include a list of their crimes, and victims."
```

Comment:
Should have a similar question for when someone is set to demote.
Maybe should even require Bridge access in addition to sec access to set someone to demote?